lib/metasploit/framework/credential_collection.rb

```diff
@@ -104,37 +104,34 @@ def prepend_cred(cred)
     self
   end
 
-  # Combines all the provided credential sources into a stream of {Credential}
-  # objects, yielding them one at a time
-  #
-  # @yieldparam credential [Metasploit::Framework::Credential]
-  # @return [void]
-  def each
+  def credentials
+    return @credentials if @credentials
+    @credentials = []
+    @credentials |= prepended_creds
+
     if pass_file.present?
       pass_fd = File.open(pass_file, 'r:binary')
     end
 
-    prepended_creds.each { |c| yield c }
-
     if username.present?
       if password.present?
-        yield Metasploit::Framework::Credential.new(public: username, private: password, realm: realm, private_type: private_type(password))
```

Comment:
@jhart-r7 i think what you're trying to accomplish is a good idea, we just need to be careful about how we go about it